Sources/SwiftDocC/Model/Identifier.swift

```diff
@@ -669,3 +670,17 @@ func urlReadableFragment<S: StringProtocol>(_ fragment: S) -> String {
     
     return fragment
 }
+
+/// Creates a more writable version of an articles file name for use in documentation links.
+///
+/// Compared to `urlReadablePath(_:)` this also removes most punctuation characters.
+/// For example, a filename like `"One, two, & three!"` is converted to `"One-two-three"` instead of `"One,-two-&-three!"`.
```

Comment:
Correcting the example is great but removing the comparison to `urlReadablePath(_:)` removed what I feel was most important information in this documentation comment; a clue at when a developer should use this function and when they should use `urlReadablePath(_:)` instead. That description was also out-of-date but I prefer to update it rather than remove it.

The real answer to when this should be used is; only for a specific purpose in `PathHierarchyBasedLinkResolver` so I moved this function there and made it file private instead.